db/schema_tables.cc

```diff
@@ -1829,8 +1829,10 @@ static future<> merge_aggregates(distributed<service::storage_proxy>& proxy, sch
             cql3::functions::functions::add_function(create_aggregate(db, *val.first, val.second));
         }
         for (const auto& val : diff.dropped) {
-            auto func = create_aggregate(db, *val.first, val.second);
-            cql3::functions::functions::remove_function(func->name(), func->arg_types());
+            cql3::functions::function_name name{
+                val.first->get_nonnull<sstring>("keyspace_name"), val.first->get_nonnull<sstring>("aggregate_name")};
+            auto arg_types = read_arg_types(db, *val.first, name.keyspace);
+            cql3::functions::functions::remove_function(name, arg_types);
```

Comment:
Maybe we should drop aggregates before we drop scalar functions? It doesn't make a huge difference but it prevents a point in time where the data structure is self-inconsistent.

Reply:
The problem here is that we the same `do_merge_schema` for a vector of mutations, no matter if it contains a dropping of a scalar function, or if it contains creation of a new aggregate.
Currently we deal with UDFs first and with UDAs later - this works for UDA creations but not for keyspace deletions, and if we switch the order, what works will be switched as well.
What would probably make most sense is deleting aggregates and then functions, then creating functions and then aggregates, but this would require a bigger change in code.

The change posted here would still be positive though, because what's changed here is that we don't perform the unnecessary parts of the `create_aggregate` call